cmd/handlers.go

```diff
@@ -89,8 +89,29 @@ func handlePingRequest(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// verifyToken checks the request headers for a valid bearer token
+func verifyToken(w http.ResponseWriter, r *http.Request) bool {
+	if serverToken == "" {
+		return true
+	}
+	reqToken := r.Header.Get("Authorization")
```

Comment:
Is this going to work as desired, with the caching in nginx & google CDN?

Reply:
I forgot about that CDN layer but thinking over it some more I think it will still work. The m3u8 files are served by the same handler and those bypass, so the auth would restrict who can fetch the playlist. Subsequent media files would then be served by the CDN.

I'll do some testing today to confirm.

Reply:
Going to revisit auth later